yt/frontends/amrvac/datfile_utils.py

```diff
@@ -151,17 +151,12 @@ def get_single_block_data(istream, byte_offset, block_shape):
 
 def get_single_block_field_data(istream, byte_offset, block_shape, field_idx):
     """retrieve a specific block (ONE field) from a datfile"""
-    istream.seek(byte_offset)
-
     # compute byte size of a single field
     field_shape = block_shape[:-1]
     fmt = ALIGN + np.prod(field_shape) * "d"
     byte_size_field = struct.calcsize(fmt)
 
-    # Read actual data
-    istream.seek(byte_size_field * field_idx, 1)  # seek forward
-    d = struct.unpack(fmt, istream.read(struct.calcsize(fmt)))
-
-    # Fortran ordering
-    block_field_data = np.reshape(d, field_shape, order="F")
-    return block_field_data
+    istream.seek(byte_offset + byte_size_field * field_idx)
```

Comment:
Reducing the number of seeks is really helpful.  It might be possible to sort the calls to `get_single_block_of_data` by `offset` and `field_idx` (even sorting by that tuple should work) to make sure we're reading them in the right order, too.

Reply:
sounds like a good idea, let's try that !

Reply:
So I was able to sort by field but it may not have a huge impact because of how AMRVAC dump files represent data (fields are on the inner loop of the writing routine, individual fields are never contiguous). I've tried sorting grids too and got no significant gain. The dataset I'm using to benchmark this also happens to use a single "data chunk", so there's no way to measure if sorting chunks would help. In conclusion I don't think there's anything I can do here.